testing/typescript/data/stores.ts

```diff
@@ -20,6 +21,17 @@ export async function infersTItemFromComplexLoadResult() {
   const value: { a: number } = result;
 }
 
+export async function loadReturnRightTypeIfLoadDefinedInOptions() {
+  const store = new CustomStore({
+    // eslint-disable-next-line @typescript-eslint/require-await
+    load: async () => ({
+      data: [{ a: 1 }],
```

Comment:
A test for custom loaded type required:
```suggestion
      data: { data: [{ ab: 1 }], extraField: 123},
```

E.g. such test would fail, because `result` is `any`:

```ts
  const store = new CustomStore({
    // eslint-disable-next-line @typescript-eslint/require-await
    load: async () => ({
      data: { data: [{ ab: 1 }], extraField: 123},
    }),
  });

  const result = await store.byKey(ANY);
  notAny(result);
```